### ISSUE TITLE
Can SerialName annotation support alternate?

### ISSUE THREAD

**Issue by chenfei0928** [Kotlin/kotlinx.serialization#203]
Can SerialName annotation support alternate?
Like Gson's SerializedName annotation.

**Comment by ToxicBakery** [comment]
I'm actually in need of this as well. One thing I don't really like about gson's is the API for it personally. I would propose, if possible, that this implementation just use a single parameter overloaded constructor like

```kotlin
@SerialName("Example")
@SerialName(listOf("Example", "example"))

// or if var args are acceptable
@SerialName("Example", "example")
```

**Comment by Zeta611** [comment]
This would be awesome!

**Comment by tjohnn** [comment]
@ToxicBakery I don't think the api you are suggesting will work well for `stringifying`, gson is using `value` and `alternate` so it can use the `value`'s value for transforming to json from an object

**Comment by ToxicBakery** [comment]
I don't see that as an issue. I'll settle for having support at all though regardless of the API :smile:

**Comment by PingForward** [comment]
Any updates on this feature? Waiting for it too.

**Comment by Vivecstel** [comment]
Any news ?

**Comment by mochadwi** [comment]
Any updates?

**Comment by mochadwi** [comment]
Edited: they're working on itu here #597

**Comment by qwwdfsad** [comment]
It will be likely supported in the upcoming releases, but in a JSON specific.
Please keep track of that issue here: https://github.com/Kotlin/kotlinx.serialization/issues/203

**Comment by lic2050** [comment]
> Can SerialName annotation support alternate?
> Like Gson's SerializedName annotation.

老哥，这个东西有了吗？

**Comment by sandwwraith** [comment]
Fixed by #1375
